### PR TITLE
Audio Descriptions

### PR DESCRIPTION
**Pull request by dadewoyin16**
# TL;DR

- Functionality for audio descriptions. 
- Using class components instead of functions/hooks to stick with current video-react structure. 
- video-react's linter made some things look weird 🤷🏾‍♂️

Let me know if I'm missing something or could be doing something more efficient.

### CODE REVIEW

File: src/components/menu/AudioDescription.js [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  constructor(props, context) {
+    super(props, context);
+
+    this.audioRef = React.createRef();
+  }
+
+  componentDidUpdate() {
+    // keep audio description in sync with video player
+    if (!this.audioRef || !this.audioRef.current) return;
+
```
Comment (AriannaLanz) [inline src/components/menu/AudioDescription.js:22]: Great comments!

File: src/components/menu/AudioDescription.js [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    if (this.props.player.currentTime > this.audioRef.current.duration) {
+      this.audioRef.current.currentTime = 0;
+      this.audioRef.current.pause();
+    }
+
+    // keep audio in sync with video
+    if (
+      Math.abs(
+        this.audioRef.current.currentTime - this.props.player.currentTime
+      ) > 0.5
```
Comment (AriannaLanz) [inline src/components/menu/AudioDescription.js:33]: What does the 0.5 represent?

**Review by AriannaLanz — APPROVED** [review]
Added a question, but otherwise LGTM!

File: src/components/menu/AudioDescription.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import classNames from 'classnames';
+
+const propTypes = {
+  player: PropTypes.object,
+  actions: PropTypes.object,
+  className: PropTypes.string,
+  audioDescriptions: PropTypes.any
```
Comment (ombene) [inline src/components/menu/AudioDescription.js:9]: Non-blocking, and I don't feel strongly about this, but have you considered  `PropTypes.arrayOf(PropTypes.shape({/* ... */}))` here? `PropTypes.any` seems rather vaguer than necessary.

File: src/components/menu/AudioDescription.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    if (this.props.player.muted) this.audioRef.current.muted = true;
+  }
+
+  render() {
+    const { player, className, actions, audioDescriptions } = this.props;
+
+    if (!player.audioDescriptions.length && audioDescriptions) {
+      const descriptions = [{ language: 'OFF' }, ...audioDescriptions];
+      actions.setAudioDescriptions(descriptions);
+    }
```
Comment (ombene) [inline src/components/menu/AudioDescription.js:53]: Non-blocking feedback. From what I'm seeing, this is the logic responsible for making `OFF` the first item in the menu and also for setting the global audio descriptions, right? I feel like state change triggers inside `render` are unexpected / non-standard, and a more natural place for this logic could be a method (e.g. `synchronizeAudioDescriptionState`) that gets called on `componentDidMount` and then also called on a change to the `audioDescriptions` coming from `props` in `componentDidUpdate`. With the added benefit that the `audioDescriptions` prop would then be allowed to change from the outside, and that change will be reflected on the inside.

File: src/components/menu/AudioDescription.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+      this.audioRef.current.currentTime = this.props.player.currentTime;
+
+    // update settings if necessary
+    if (this.audioRef.current.volume !== this.props.player.volume)
+      this.audioRef.current.volume = this.props.player.volume;
+    if (this.props.player.paused && !this.audioRef.current.paused)
+      this.audioRef.current.pause();
+    if (!this.props.player.paused && this.audioRef.current.paused)
+      this.audioRef.current.play();
+    if (this.props.player.muted) this.audioRef.current.muted = true;
```
Comment (ombene) [inline src/components/menu/AudioDescription.js:44]: Wow, nice job on the details here. I would've forgotten to include at least half of these.

**Review by ombene — APPROVED** [review]
Looking sharp. This is so cool! One note of non-blocking feedback below.

**Pull request merged**